Patch developed by Kurt Lust to ensure that the proper compiler wrappers are used to
generate the opari2 executable. Otherwise the system gcc is used.
This patch is for cpeGNU, cpeCray and cpeAOCC only. Currently OPARI2 does not compile
correctly with the wrappers in cpeAMD.
diff -Nrup tau-2.33.2/utils/include/Makefile.skel.orig tau-2.33.2/utils/include/Makefile.skel
--- tau-2.33.2/utils/include/Makefile.skel.orig	2024-07-30 19:28:20.000000000 +0300
+++ tau-2.33.2/utils/include/Makefile.skel	2024-07-30 19:28:40.000000000 +0300
@@ -227,7 +227,7 @@ INSTALL_OTF_TOOLS = $(INSTALL_EBS2OTF)
 #CRAYCNL#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=intel CXXFLAGS_FOR_BUILD= CC_FOR_BUILD=/usr/bin/gcc CXX_FOR_BUILD=/usr/bin/g++ FC_FOR_BUILD=ifort FFLAGS_FOR_BUILD= FC=ifort FCFLAGS_FOR_BUILD= LIBS_FOR_BUILD= F77=ifort MPIF77=ifort CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD= LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared ac_scorep_platform=cray ac_scorep_platform_data_provided=yes ac_scorep_cross_compiling=yes --cache-file=/dev/null --srcdir=.. #ENDIF#
 #PGI_FORTRAN#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=pgi CXXFLAGS_FOR_BUILD= CC_FOR_BUILD=/usr/bin/gcc CXX_FOR_BUILD=/usr/bin/g++ FC_FOR_BUILD=pgfortran FFLAGS_FOR_BUILD= FC=pgfortran FCFLAGS_FOR_BUILD= LIBS_FOR_BUILD= F77=pgfortran MPIF77=pgfortran CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD= LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared ac_scorep_platform=cray ac_scorep_platform_data_provided=yes ac_scorep_cross_compiling=yes --cache-file=/dev/null --srcdir=.. #ENDIF#
 #GNU_GFORTRAN#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=gcc CXXFLAGS_FOR_BUILD= CC_FOR_BUILD=/usr/bin/gcc CXX_FOR_BUILD=/usr/bin/g++ FC_FOR_BUILD=gfortran FFLAGS_FOR_BUILD= FC=gfortran FCFLAGS_FOR_BUILD= LIBS_FOR_BUILD= F77=gfortran MPIF77=gfortran CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD= LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared ac_scorep_platform=cray ac_scorep_platform_data_provided=yes ac_scorep_cross_compiling=yes --cache-file=/dev/null --srcdir=.. #ENDIF#
-#CRAY_FORTRAN#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=cray CXXFLAGS_FOR_BUILD= CC_FOR_BUILD=/usr/bin/gcc CXX_FOR_BUILD=/usr/bin/g++ FC_FOR_BUILD=ftn FFLAGS_FOR_BUILD= FC=ftn FCFLAGS_FOR_BUILD= LIBS_FOR_BUILD= F77=ftn MPIF77=ftn CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD= LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared ac_scorep_platform=cray ac_scorep_platform_data_provided=yes ac_scorep_cross_compiling=yes --cache-file=/dev/null --srcdir=.. #ENDIF#
+#CRAY_FORTRAN#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=cray CXXFLAGS_FOR_BUILD="-O2 -g" CC_FOR_BUILD=cc CXX_FOR_BUILD=CC FC_FOR_BUILD=ftn FFLAGS_FOR_BUILD="-O2 -g" FC=ftn FCFLAGS_FOR_BUILD="-O2 -g" LIBS_FOR_BUILD= F77=ftn MPIF77=ftn CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD="-O2 -g" LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared ac_scorep_platform=cray ac_scorep_platform_data_provided=yes ac_scorep_cross_compiling=yes --cache-file=/dev/null --srcdir=.. #ENDIF#
 #TAU_CRAY_AARCH64#CRAY_OPARI2_CONFIG= CC_FOR_BUILD=cc CXX_FOR_BUILD=CC FC_FOR_BUILD=ftn CC=cc CXX=CC FC=ftn F77_FOR_BUILD=ftn F77=ftn #ENDIF#
 #TAU_CRAY_AARCH64#CRAY_OPARI2_CONFIG_CRAYCNL= --with-compiler-suite=gnu CXXFLAGS_FOR_BUILD= CC_FOR_BUILD=/usr/bin/gcc CXX_FOR_BUILD=/usr/bin/g++ FC_FOR_BUILD=ifort FFLAGS_FOR_BUILD= FC=ifort FCFLAGS_FOR_BUILD= LIBS_FOR_BUILD= F77=ifort MPIF77=ifort CC=cc CXX=CC MPICXX=CC MPIFC=ftn CPPFLAGS_FOR_BUILD= CFLAGS_FOR_BUILD= LDFLAGS_FOR_BUILD= MPICC=cc --enable-shared --cache-file=/dev/null --srcdir=.. #ENDIF#
 #INTELIFORT#CRAY_OPARI2_CONFIG=$(CRAY_OPARI2_CONFIG_CRAYCNL) #ENDIF#
